neural_compressor/torch/quantization/load_entry.py

```diff
@@ -25,43 +25,87 @@
     RTNConfig,
     TEQConfig,
 )
+from neural_compressor.torch.utils import LoadFormat
 
 config_name_mapping = {
     FP8_QUANT: FP8Config,
 }
 
 
-def load(output_dir="./saved_results", model=None):
-    """The main entry of load for all algorithms.
+def load(model_name_or_path, original_model=None, format="default", device="cpu", *model_args, **kwargs):
+    """Load quantized model.
 
-    Args:
-        output_dir (str, optional): path to quantized model folder. Defaults to "./saved_results".
-        model (torch.nn.Module, optional): original model, suggest to use empty tensor.
+    1. Load INC quantized model in local.
+    2. Load HuggingFace quantized model, including GPTQ/AWQ models and upstreamed INC quantized models in HF model hub.
+
+    case 1: WOQ
+        # huggingface model
+        from neural_compressor.torch.quantization import load
+        load(model_name_or_path=model_name_or_path)
+
+        # local model
+        from neural_compressor.torch.quantization import load
+        load(model_name_or_path="saved_results", original_model=fp32_model)
+
+    case 2: INT8/FP8
+        from neural_compressor.torch.quantization import load
+        load(model_name_or_path='saved_result', original_model=fp32_model)
+
+    case 3: TorchScript (IPEX)
+        from neural_compressor.torch.quantization import load
+        load(model_name_or_path='saved_result')
 
+    Args:
+        model_name_or_path (str):  torch checkpoint directory or hugginface model_name_or_path.
+            If 'format' is set to 'huggingface', it means the huggingface model_name_or_path.
+            If 'format' is set to 'default', it means the 'checkpoint_dir'.
+            Parameter should not be None. it coworks with 'original_model' parameter to load INC
+            quantized model in local.
+        original_model (torch.nn.module, optional): original model before quantization.
+            Needed if 'format' is set to 'default' and not TorchScript model.Defaults to None.
+        format (str, optional): 'defult' for loading INC quantized model.
+            'huggingface' for loading huggingface WOQ causal language model. Defaults to "default".
+        device (str, optional): 'cpu', 'hpu' or 'cuda'. specify the device the model will be loaded to.
+        model_args (sequence of positional arguments, optional):
+            all remaining positional arguments for loading huggingface models.
+            Will be passed to the huggingface model's `__init__` method.
+        kwargs (remaining dictionary of keyword arguments, optional):
+            remaining dictionary of keyword arguments for loading huggingface models.
+            Will be passed to the huggingface model's `__init__` method, such as 'trust_remote_code', 'revision'.
     Returns:
-        The quantized model
+        torch.nn.Module: quantized model
```

Comment:
not only nn.Module. includes TorchScript model with IPEX and fx graph with pt2e.

Reply:
updated in https://github.com/intel/neural-compressor/pull/1826/commits/939a261dd4554ea6e45dc2ae23586ab7eda4ed68